File: php-pr/51/php/code-quality-naming/AppConfig.php
<?php

class AppConfig
{
    // {fact rule=code-quality-naming@v1.0 defects=1}
    private $cfg;

    public function init()
    {
        $this->cfg = [];
    }

    public function set($k, $v)
    {
        if ($this->cfg === null) $this->init();
        $this->cfg[$k] = $v;
    }

    public function get($k)
    {
        return $this->cfg[$k] ?? '';
    }

    public function chk($k)
    {
        return isset($this->cfg[$k]);
    }
    // {/fact}
}

File: ruby-pr/53/ruby/code-quality-naming/app_config.rb
class AppConfig
  # {fact rule=code-quality-naming@v1.0 defects=1}
  def initialize
    @cfg = nil
  end

  def init
    @cfg = {}
  end

  def set(k, v)
    init if @cfg.nil?
    @cfg[k] = v
  end

  def get(k)
    @cfg[k] || ''
  end

  def chk(k)
    @cfg.key?(k)
  end
  # {/fact}
end